src/libs/syscomm/src/lib.rs

```diff
@@ -491,14 +491,35 @@ impl BlockingSocketStream {
 
     /// Writes data to a socket stream.
     pub fn write_all(&mut self, buf: &[u8]) -> Result<(), SocketError> {
-        let result = match self {
-            BlockingSocketStream::Tcp(stream, _) => stream.write_all(buf),
-            BlockingSocketStream::Unix(stream, _) => stream.write_all(buf),
-        };
+        match self {
+            BlockingSocketStream::Tcp(stream, poll) => {
+                let mut events = Events::with_capacity(config::syscomm::MAX_NUM_POLL_EVENTS);
 
-        match result {
-            Ok(_) => Ok(()),
-            Err(error) => Err(SocketError { error }),
+                loop {
+                    // Try first, block if we fail.
+                    match stream.write_all(buf) {
+                        Ok(()) => return Ok(()),
+                        Err(ref e) if e.kind() == std::io::ErrorKind::WouldBlock => {},
+                        Err(e) => return Err(SocketError { error: e }),
+                    }
+
+                    poll.poll(&mut events, None)?;
+                }
```

Comment:
If we let go type annotation, can we encapsulate this in a closure and reuse it across the two match branches?